### PR TITLE
In Project Summary sidebar, order by most recently viewed projects

### PR DESCRIPTION
**Pull request by dtaylor113**
DO NOT MERGE until origin-web-common's version is bumped!

This is the second PR towards resolving issue [console:#1895](https://github.com/openshift/origin-web-console/issues/1895)

ProjectSummary updated to use RecentlyViewedProjects.orderByMostRecentlyViewed(..).  CreateFromBuilder and OrderService wizards updated to use ProjectServices.create(...).

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "origin-web-catalog",
-  "version": "0.0.34",
+  "version": "0.0.35",
```
Comment (jeff-phillips-18) [inline bower.json]: We  shouldn't bump the version as part of a separate PR. Bumping the version should be its own PR to avoid conflicts and version churn.

Comment (dtaylor113) [inline bower.json]: Ok, going forward I will do separate PRs for version bumps.

Comment (jeff-phillips-18) [inline bower.json]: I would still suggest removing it from the PR. If/when it is merged, a separate PR to bump the version can be added and resolved quickly.

Issue being, if we want to bump w/o waiting on the PR or want to hold the version bump for additional PRs, then we can if we do not bump within this PR.

File: app/mockServices/mockProjects.service.ts [hunk 2]
```diff
@@ -49,4 +49,16 @@ export class ProjectsService implements IProjectsService {
     return deferred.promise;
   }
 
+  public create(name: string, displayName: string, desc: string) {
+    let deferred = this.$q.defer();
+    var project: any = {};
```
Comment (spadgett) [inline app/mockServices/mockProjects.service.ts]: Let's return something like

```js
let project: any = {
  apiVersion: "v1",
  kind: "Project",
  metadata: {
    name: name,
    annotations: {}
  }
}

if (displayName) {
  project.metadata.annotations["openshift.io/display-name"] = displayName;
}

if (desc) {
  project.metadata.annotations["openshift.io/description"] = desc;
}
```

File: src/components/order-service/order-service.controller.ts [hunk 3]
```diff
@@ -202,19 +202,10 @@ export class OrderServiceController implements angular.IController {
...
-      };
-      this.DataService
-        .create('projectrequests', null, projReqObj, this.$scope)
-        .then( (data: any) => {
-          this.ctrl.selectedProject = data;
+      this.ProjectsService
+        .create(newProjName, newProjDisplayName, newProjDesc)
+        .then( (project: any) => {
+          this.ctrl.selectedProject = project;
           this.ctrl.projectDisplayName = this.$filter('displayName')(this.ctrl.selectedProject);
```
Comment (spadgett) [inline src/components/order-service/order-service.controller.ts]: Not new, but let's pass `project` here instead of `ctrl.selectedProject`. The project request template could potentially change the display name, so better to use what we get back.

File: src/components/projects-summary/projects-summary.controller.ts [hunk 4]
```diff
@@ -31,7 +32,8 @@ export class ProjectsSummaryController implements angular.IController {
   private AuthService: any;
   private DataService: any;
   private Constants: any;
-  private RecentlyViewed: any;
+  private RecentlyViewedItems: any;
+  private RecentlyViewedProjects: any;
```
Comment (spadgett) [inline src/components/projects-summary/projects-summary.controller.ts]: Nit: I'd rather make it match the service name exactly, so `RecentlyViewedProjectsService` here and below.

File: src/components/projects-summary/projects-summary.controller.ts [hunk 5]
```diff
@@ -136,6 +140,7 @@ export class ProjectsSummaryController implements angular.IController {
     var projects: any = _.toArray(projectData.by('metadata.creationTimestamp'));
     var orderByDate: any = this.$filter('orderObjectsByDate');
     this.ctrl.projects = orderByDate(projects, true);
+    this.ctrl.projects = this.RecentlyViewedProjects.orderByMostRecentlyViewed(this.ctrl.projects);
```
Comment (spadgett) [inline src/components/projects-summary/projects-summary.controller.ts]: Per my comment in the common PR, would just call `orderByMostRecentlyViewed` here and do the fallback ordering by creation date inside of `orderByMostRecentlyViewed`.

File: app/mockServices/mockProjects.service.ts (hunk 2 above)
Comment (dtaylor113) [inline app/mockServices/mockProjects.service.ts]: Fixed

File: src/components/order-service/order-service.controller.ts (hunk 3 above)
Comment (dtaylor113) [inline src/components/order-service/order-service.controller.ts]: Fixed

File: src/components/projects-summary/projects-summary.controller.ts (hunk 4 above)
Comment (dtaylor113) [inline src/components/projects-summary/projects-summary.controller.ts]: Fixed

File: src/components/projects-summary/projects-summary.controller.ts (hunk 5 above)
Comment (dtaylor113) [inline src/components/projects-summary/projects-summary.controller.ts]: Fixed

**Comment by dtaylor113** [comment]
Removed the version bump from this PR

File: src/components/projects-summary/projects-summary.controller.ts [hunk 6]
```diff
@@ -14,6 +14,7 @@ export class ProjectsSummaryController implements angular.IController {
     'DataService',
     'Logger',
     'ProjectsService',
+    'RecentlyViewedProjectsService',
     'RecentlyViewedServiceItems'
   ];
```
Comment (jeff-phillips-18) [inline src/components/projects-summary/projects-summary.controller.ts]: $filter is no longer used.

Comment (jeff-phillips-18) [inline src/components/projects-summary/projects-summary.controller.ts]: $element and Catalog are not used (not this PR).

Comment (dtaylor113) [inline src/components/projects-summary/projects-summary.controller.ts]: Removed those unused injections.  Fixed

**Comment by jeff-phillips-18** [comment]
Bah! Did not mean to hit merge.

**Comment by spadgett** [comment]
Probably easiest to just bump common in a follow on now that this is merged
